### PR TITLE
docker: update Dockerfiles to use focal images

### LINKED ISSUE #382: Docker images still on bionic, old versions of glvnd don't seem to work with latest versions of nvidia-docker2
I was trying to get ignition working via docker (latest version, with nvidia-container-toolkit installed) and I noted that it is still on bionic, which I believe may be interfering with the new nvidia docker support (which doesn't require nvidia-docker2, but nvidia-container-toolkit). 
https://github.com/ignitionrobotics/ign-gazebo/blob/master/docker/Dockerfile.ignition#L7

I am able to get the following to work (please note, the script you include on dockerhub is what I have named `docker-gpu.sh`.
```
./docker-gpu.sh nvidia/opengl:1.2-glvnd-runtime
```
and then installing mesa-utils and running glxgears. I was not able to get glxgears working with your image https://hub.docker.com/r/ignitionrobotics/ignition using the dome tag.

The error messages are your standard:
```
# LIBGL_DEBUG=verbose glxgears
libGL: screen 0 does not appear to be DRI2 capable
libGL: MESA-LOADER: dlopen(/usr/lib/x86_64-linux-gnu/dri/swrast_dri.so)
libGL: Can't open configuration file /etc/drirc: No such file or directory.
libGL: Can't open configuration file /root/.drirc: No such file or directory.
libGL: Can't open configuration file /etc/drirc: No such file or directory.
libGL: Can't open configuration file /root/.drirc: No such file or directory.
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
X Error of failed request:  GLXBadContext
  Major opcode of failed request:  151 (GLX)
  Minor opcode of failed request:  6 (X_GLXIsDirect)
  Serial number of failed request:  43
  Current serial number in output stream:  42
```
I believe this can be fixed by updating to the focal images.

### PR DESCRIPTION
**Pull request by russkel**
To close #382 

  * use focal base images instead of bionic, including using newer version of libglvnd
  * runtime=nvidia is deprecated, gpus=all has replaced it in newer
  versions of docker

This has not been built or tested by me.

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -154,7 +154,7 @@ Docker has two available versions: Community Edition (CE) and Enterprise Edition
 
 1. Verify the installation:
 
-        docker run --runtime=nvidia --rm nvidia/cuda nvidia-smi
```
Comment (j-rivero) [inline docker/README.md:157]: If I understand it correctly, this change is not strictly related to the update to focal but more about the version of docker in the user system. I think we can update it as you did in the PR since modern docker uses the native nvidia integration.

**Review by j-rivero — COMMENTED** [review]
I tried the changes using the new Dome release. Everything seems fine except that the ignition-dome package is not yet released, should be ready tomorrow. Let's wait 48 hours to give this a final run. Thanks @russkel for the changes!

File: docker/README.md (hunk 1 above)
Comment (russkel) [reply docker/README.md:157]: This is correct. I decided to update to reflect the docs found at https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#docker to hopefully reduce confusion.

I'm not too familiar with what is actually going on in docker/nvidia-runtime internals so maybe this change in CLI arguments is required.

**Comment by j-rivero** [comment]
> Everything seems fine except that the ignition-dome package is not yet released, should be ready tomorrow

@russkel now that dome in released, we are fine to merge these changes. I would need a couple of things from you: could you please rebase/merge the changes on `ign-gazebo4` with your branch? While doing that we need to address the DCO bot failure in the CI, there are instrcutions in the "Details" link https://github.com/ignitionrobotics/ign-gazebo/pull/388/checks?check_run_id=1181732463?

**Comment by russkel** [comment]
Hi @j-rivero, done as you have asked.

**Review by j-rivero — APPROVED** [review]
thanks you very much! ready to go.
